src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -52,24 +52,35 @@ public override void CopyFile(string sourceFullPath, string destFullPath, bool o
 
         public override void MoveFile(string sourceFullPath, string destFullPath)
         {
-            while (Interop.libc.rename(sourceFullPath, destFullPath) < 0)
+            while (Interop.libc.link(sourceFullPath, destFullPath) < 0)
             {
                 int errno = Marshal.GetLastWin32Error();
                 if (errno == Interop.Errors.EINTR) // interrupted; try again
                 {
                     continue;
                 }
-                else if (errno == Interop.Errors.EXDEV) // rename fails across devices / mount points
+                else if (errno == Interop.Errors.EXDEV) // link fails across devices / mount points
                 {
                     CopyFile(sourceFullPath, destFullPath, overwrite: false);
-                    DeleteFile(sourceFullPath);
                     break;
                 }
+                else if (errno == Interop.Errors.ENOENT && !Directory.Exists(Path.GetDirectoryName(destFullPath))) // The destFile parent directory can't be found
```

Comment:
Small issue here with the way the desktop implementation treats errors in File.Move as per #2537. File.Move is currently a special case and introducing a check for ```sourceFullPath``` here will bring about an inconsistency with the Windows implementation since ```File.Move(Path.Combine(TestDirectory, "doesntexist", "filename"), "filename2");``` would throw DirectoryNotFoundException on Unix and FileNotFoundException on Windows.

Unless we can get this changed for Desktop as well, we should probably leave the behavior as it is now i.e. not check if the parent exists for ```sourceFullPath```.